### PR TITLE
Watermark image api

### PR DESCRIPTION
**Pull request by leonkunert**
Should make #199 obsolete.
- Added limit reader
- Added some more error hadling

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -101,6 +101,7 @@ func NewServerMux(o ServerOptions) http.Handler {
 	mux.Handle(join(o, "/zoom"), image(Zoom))
 	mux.Handle(join(o, "/convert"), image(Convert))
 	mux.Handle(join(o, "/watermark"), image(Watermark))
+	mux.Handle(join(o, "/watermarkImage"), image(WatermarkImage))
```
[the remark this answers is not in the document]
Comment (h2non) [reply server.go]: Yeah, makes sense. We should use a dash or lower case all the path.

[the remark this answers is not in the document]
Comment (leonkunert) [reply server.go]: Will do.

**Comment by alewin** [comment]
any docs?

**Comment by suntong** [comment]
> any docs?

+1

**Comment by suntong** [comment]
> any docs?

+1

@alewin, have you figured out how to use it please?

**Pull request merged**